backend/src/main.py

```diff
@@ -0,0 +1,196 @@
+import argparse
+import json
+import pathlib
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from http import HTTPStatus as status
+
+import cassiopeia as cass
+import dacite
+import flask
+import flask.views as views
+import flask_cors
+import sqlalchemy
+import sqlalchemy.orm
+
+import models
+
+# setup flask & sqlalchemy
+app = flask.Flask(__name__)
+engine = sqlalchemy.create_engine("sqlite:///data.db")
+flask_cors.CORS(app)  # get rid of cors
+
+# setup cassiopeia
+cass_config = cass.get_default_config()
+cass_config["logging"]["print_calls"] = False
+cass.apply_settings(cass_config)
+
+# other config
+STATIC_PATH = "static"
+ICONS_PATH = f"{STATIC_PATH}/icons"
+
+
+class UserListView(views.MethodView):
+    def get(self):
+        """ Returns the list of all users. """
+        with sqlalchemy.orm.Session(engine) as session:
+            users = session.query(models.User).all()
+            result = [user.to_dict() for user in users]
+            return result
+
+    def post(self):
+        """ Creates a new user. """
+        data = flask.request.json
+        with sqlalchemy.orm.Session(engine) as session:
+            entity = models.User(**data)
+            session.add(entity)
+            session.commit()
+            return "", status.CREATED
+
+
+class UserDetailView(views.MethodView):
+    wait_time = timedelta(hours=1)      # time to wait between profile updates
+
+    def _get_user_data(self, id: int) -> dict | None:
+        """ Retrieves the user dict representation by id. If no user was found it will return None.
+
+        This method will update at regular intervals of time, defined by `wait_time`, the user's profile
+
+        Possible updates:
+            - profile icon: if it does not exist it will be fetched and stored
+            - last match: it is updated with the match start datetime + user's time played
+
+        Args:
+            id (int): id of the user
+
+        Returns: a dictionary containing user data if the user was found else None
+        """
+        with sqlalchemy.orm.Session(engine) as session:
+            user: models.User = session.query(models.User).filter_by(id=id).one_or_none()
+
+            # user not found
+            if user is None:
+                return None
+
+            profile: models.Profile = user.profile
+
+            # check whether profile information is out of date (or missing),
+            # and if it is, update it
+            now = datetime.now()
+            if (
+                profile.last_match_updated is None or
```

Comment:
**nitpick incoming:**
cred ca e mai bine daca pui operatorii la inceputul linei, adica de exemplu aici sa fie
```python
if (
     profile.last_match_updated is None
  or now - profile.last_match_updated >= self.wait_time
):
  # bla
```
mi se pare ca in acest mod se citeste mai usor codul